docs/DeveloperGuide.adoc

```diff
@@ -742,30 +790,282 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+Use case ends
+
+*Extensions*
+
+[none]
+* 2a. The backup file cannot be read during restoration
++
+[none]
+** 2a1. AddressBook shows an error message
++
+Use case resumes at step 3
+
+* 3a. The specified location cannot be written to during backup
+** 3a1. AddressBook shows an error message
++
+Use case resumes at step 1
+
+[discrete]
+=== Use case: Add multiple phone numbers
 
 *MSS*
 
 1.  User requests to list persons
 2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+3.  User requests to add multiple phone numbers to a specific person in the list
+4.  AddressBook add multiple phone numbers of the person
+
+*Extensions*
+
+* 3a. The number added is already existed
+** 3a1. AddressBook shows an error message
 +
-Use case ends.
+Use case ends
+
+[discrete]
+=== Use case: Search contacts by multiple criteria
+
+*MSS*
+
+1.  User requests to list contacts by specifying substrings of name and/or address
+2.  AddressBook displays the matching contacts in the GUI interface (if any)
+
+*Extensions*
+
+* 2a. The list is empty
++
+Use case ends
+
+[discrete]
+=== Use case: Sort persons names in alphabetical order
+
+*MSS*
+
+1. User requests to list persons
+2. AddressBook shows a list of persons unsorted
+3. User requests to sort persons in alphabetical order
 
 *Extensions*
 
 [none]
-* 2a. The list is empty.
+* 2a. The list is sorted
 +
-Use case ends.
+Use case ends
+
+* 2b. The list is empty
++
+Use case ends
+
+[discrete]
+=== Use case: Auto-correct misspelt words
+
+*MSS*
+
+1.  User inputs a misspelt word
+2.  AddressBook identifies and autocorrects the word
++
+Use case ends
+
+[discrete]
+=== Use case: Sort person by tag
+
+*MSS*
+
+1. User requests to sort persons with a specific tag
+2. AddressBook shows a list of persons with the same tag
++
+Use case ends
+
+*Extensions*
+
+* 1a. The tag is not recognizable
+** 1a1. Addressbook requests the user to input a new tag name
++
+Steps 1a - 1a1 are repeated util the tag name entered is correct
++
+Use case resumes from step 2
+
+* 2a. The list is empty
++
+Use case ends
+
+[discrete]
+=== Use case: Search for people who are using address book
+
+*MSS*
 
-* 3a. The given index is invalid.
+1.  User requests to search for people who are using address book
+2.  AddressBook shows a list of matching persons (if any)
 +
+Use case ends
+
+*Extensions*
+
 [none]
-** 3a1. AddressBook shows an error message.
+* 2a. The list is empty
++
+Use case ends
+
+[discrete]
+=== Use case: Delete many contacts simultaneously
+
+*MSS*
+
+1.  User requests to delete contacts with a certain tag
+2.  AddressBook deletes the contacts with the stated tag
+3.  Addressbook reports the entries that have been deleted (if any)
++
+Use case ends
+
+[discrete]
+=== Use case: Load more than 100,000 persons at one time
+
+*MSS*
+
+1.  User requests to list more than 100,000 persons at one time
+2.  AddressBook shows a list of matching persons
```

Comment:
not only matching persons. it should list 100000 persons as per the request

Reply:
@zhangshuoyang